src/go/rpk/pkg/cli/cmd/container/common/common.go

```diff
@@ -314,9 +319,23 @@ func CreateNode(
 	}, nil
 }
 
-func PullImage(c Client) error {
+func PullImage(c Client, image string) error {
+	img, err := docker.Parse(image)
+	if err != nil {
+		log.Debug(err)
+		return fmt.Errorf("invalid container image reference: %s", image)
+	}
+	named, isNamed := img.(docker.Named)
+	if !isNamed {
+		return fmt.Errorf("invalid unnamed container image reference: %s", image)
+	}
+	if docker.Domain(named) == "" {
+		// If the reference doesn't specify a registry, assume docker.io
+		image = fmt.Sprintf("%s/%s", registry, image)
+	}
```

Comment:
since this is already done by the docker engine/client, it might be simpler to identify these problems from the errors returned by the `containerStart()`

Reply:
I will try that. Sometimes in Go it's hard to check that an error is X or Y, that's why I went this way.

Reply:
Done!